lib/jxl/modular/transform/enc_palette.cc

```diff
@@ -95,9 +96,69 @@ static int QuantizeColorToImplicitPaletteIndex(
 
 }  // namespace palette_internal
 
-Status FwdPalette(Image &input, uint32_t begin_c, uint32_t end_c,
-                  uint32_t &nb_colors, bool ordered, bool lossy,
-                  Predictor &predictor, const weighted::Header &wp_header) {
+int RoundInt(int value, int div) {  // symmetric rounding around 0
+  if (value < 0) return -RoundInt(-value, div);
+  return (value + div / 2) / div;
+}
+
+struct PaletteIterationData {
+  static constexpr int kMaxDeltas = 128;
+  bool final_run = false;
+  std::vector<pixel_type> deltas[3];
```

Comment:
Can we do `std::array<std::vector<...>, 3>` instead? (I'm not sure it changes anything, but ...)

Reply:
Jyrki had suggested this for consistency, I will leave it like this.